plugins/catalog-backend-module-ldap/src/ldap/read.ts

```diff
@@ -479,18 +479,21 @@ export function resolveRelations(
     }
   }
   for (const [groupN, membersN] of groupMember.entries()) {
-    const group = groupMap.get(groupN);
+    const group = getValueFromMapWithInsensitiveKey(groupMap, groupN);
     if (group) {
       for (const memberN of membersN) {
         // Group members can be both users and groups in the input model, so
         // try both
-        const memberUser = userMap.get(memberN);
+        const memberUser = getValueFromMapWithInsensitiveKey(userMap, memberN);
         if (memberUser) {
           ensureItems(newUserMemberOf, stringifyEntityRef(memberUser), [
             stringifyEntityRef(group),
           ]);
         } else {
-          const memberGroup = groupMap.get(memberN);
+          const memberGroup = getValueFromMapWithInsensitiveKey(
+            groupMap,
+            memberN,
+          );
```

Comment:
Why don't we add an additional line `groupMap.set(group.metadata.annotations![LDAP_DN_ANNOTATION]?.toLocaleLowerCase('en-US'), group);` on line 428 above, and look up via the lowercased name here? Not fond of the hard iteration, there can be a hundred thousand users in here and that will be 100000*100000 loop rounds which will absolutely crush your process

Reply:
Just realised you said **additional**, will do it based off that and clean it up based off my assumption. Happy to change it if decide different.

Reply:
Can someone re-run the E2E 20 action? Not sure why it's failing, my testing was all on Node 20

Reply:
Not your fault, a third party library made a broken release on npm. A rebase on latest master may help

Reply:
Yup, all good now. I've manually copied the code into my on-prem installation and its working as expected, as well as unit tests all covered. Thanks for your help, was in a bit of a rush early on to get something functioning internally but had more time to look into it since then as we discussed earlier.